roles/ssh_jumper/molecule/default/converge.yml

```diff
@@ -0,0 +1,146 @@
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Converge
+  hosts: instance
+  vars:
+    cifmw_basedir: "/opt/basedir"
+  tasks:
+    - name: Crate SSH keypair
+      register: _test_key
+      community.crypto.openssh_keypair:
+        comment: "test-key"
+        path: "{{ (cifmw_basedir, 'ssh/id_test') | path_join }}"
+        type: "ecdsa"
+
+    - name: Add SSH jumper with proxy
+      vars:
+        cifmw_ssh_jumper_config:
+          ssh_dir: "{{ cifmw_basedir }}/ssh"
+          hostname: '192.168.250.10'
+          proxy_host: "proxy.example.com"
+          proxy_user: cloud-user
+          patterns:
+            - test
+            - test.node
+      ansible.builtin.include_role:
+        name: ssh_jumper
+
+    - name: Add SSH host - minimal config
+      vars:
+        cifmw_ssh_jumper_config:
+          hostname: minimal.example.com
+          ssh_dir: "{{ cifmw_basedir }}/ssh"
+      ansible.builtin.include_role:
+        name: ssh_jumper
+
+    - name: Add SSH jumper without proxy
+      vars:
+        cifmw_ssh_jumper_config:
+          target: instance
+          ssh_dir: "{{ cifmw_basedir }}/ssh"
+          hostname: '192.168.250.11'
+          identity_file: "{{ _test_key.filename }}"
+          patterns:
+            - test
+            - test.node
+      ansible.builtin.include_role:
+        name: ssh_jumper
+
+    - name: Slurp ssh/config
+      register: ssh_config
+      ansible.builtin.slurp:
+        src: "{{ cifmw_basedir }}/ssh/config"
+
+    - name: Slurp ssh/cifw_ssh_config.d/cifmw-minimal.example.com.conf
+      register: cifmw_minimal_example_com
+      ansible.builtin.slurp:
+        src: "{{ cifmw_basedir }}/ssh/cifw_ssh_config.d//cifmw-minimal.example.com.conf"
+
+    - name: Slurp ssh/cifw_ssh_config.d/cifmw-192.168.250.10.conf
+      register: cifw_ssh_config_d_192_168_250_10
+      ansible.builtin.slurp:
+        src: "{{ cifmw_basedir }}/ssh/cifw_ssh_config.d//cifmw-192.168.250.10.conf"
+
+    - name: Slurp ssh/cifw_ssh_config.d/cifmw-192.168.250.11.conf
+      register: cifw_ssh_config_d_192_168_250_11
+      ansible.builtin.slurp:
+        src: "{{ cifmw_basedir }}/ssh/cifw_ssh_config.d//cifmw-192.168.250.11.conf"
+
+    - name: Assert Inlcude in ssh_config
+      vars:
+        _ref: |
+          Include cifw_ssh_config.d/*.conf
+        _res: "{{ ssh_config['content'] | b64decode }}"
+      ansible.builtin.assert:
+        that:
+          - _ref == _res
+
+    - name: Assert SSH host - minimal config
+      vars:
+        _ref: |
+          Host minimal.example.com
+              Hostname minimal.example.com
```

Comment:
Manual page ssh_config(5) says `Hostname`.

```
       Hostname
               Specifies  the  real host name to log into.  This can be used to specify nicknames or abbreviations for hosts.  Arguments to Hostname accept the tokens described in the “TOKENS” section.  Numeric IP addresses are also
               permitted (both on the command line and in Hostname specifications).  The default is the name given on the command line.
```